src/index.ts

```diff
@@ -139,34 +141,74 @@ class Index<T> implements Types.IndexInterface<T> {
   /// INDEX
   ///
   /**
-   * Show index information.
+   * Get index information.
    * @memberof Index
-   * @method show
+   * @method getInfo
    */
-  async show(): Promise<Types.IndexResponse> {
+  async getInfo(): Promise<Types.IndexResponse> {
     const url = `/indexes/${this.uid}`
 
-    return await this.httpRequest.get<Types.IndexResponse>(url)
+    const res = await this.httpRequest.get<Types.IndexResponse>(url)
+    this.primaryKey = res.primaryKey
+    return res
+  }
+
+  /**
+   * Fetch and update Index information.
+   * @memberof Index
+   * @method fetchInfo
+   */
+  async fetchInfo(): Promise<this> {
+    await this.getInfo()
+    return this
+  }
+
+  /**
+   * Get Primary Key.
+   * @memberof Index
+   * @method fetchPrimaryKey
+   */
+  async fetchPrimaryKey(): Promise<string | undefined> {
+    this.primaryKey = (await this.getInfo()).primaryKey
+    return this.primaryKey
+  }
+
+  /**
+   * Create an index.
+   * @memberof Index
+   * @method create
+   */
+  static async create<T = any>(
+    config: Types.Config,
+    uid: string,
+    options: Types.IndexOptions = {}
+  ): Promise<Index<T>> {
+    const url = '/indexes'
+
+    const req = new HttpRequests(config)
+    const index = await req.post(url, { ...options, uid })
+    return new Index(config, uid, index.primaryKey)
   }
 
   /**
    * Update an index.
    * @memberof Index
-   * @method updateIndex
+   * @method update
    */
-  async updateIndex(data: Types.IndexOptions): Promise<Types.IndexResponse> {
+  async update(data: Types.IndexOptions): Promise<Types.IndexResponse> {
     const url = `/indexes/${this.uid}`
 
-    return await this.httpRequest.put(url, data)
+    const index = await this.httpRequest.put(url, data)
+    this.primaryKey = index.primaryKey
+    return index
```

Comment:
Could you return the updated `Index` object instead of the JSON here? If you think this is not related to this PR I can open an issue to make the `update` method return an `Index` object instead of a JSON object.